spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/NNPrimitive.scala

```diff
@@ -353,6 +353,194 @@ object NNPrimitive {
     }
   }
 
+  def im2colDoubleNHWC(
+    fInput: Tensor[Double], input: Tensor[Double],
+    kW: Int, kH: Int,
+    dW: Int, dH: Int,
+    padLeft: Int, padTop: Int,
```

Comment:
padRight and padBottom can be determined by input and outputsize (padBottom = outputHeight - inputHeight - padTop)